stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -316,23 +319,32 @@ def _get_first_picking(self, raise_if_not_found=False):
             for device in self.stock_device_type_ids:
                 device_domains.append(self._get_picking_domain_for_device(device))
             domain = AND([domain, OR(device_domains)])
-        picking = self._execute_search_pickings(domain, limit=1)
-        if not picking and not no_limit and raise_if_not_found:
-            self._raise_create_batch_not_possible()
+            picking = self._execute_search_pickings(domain, limit=1)
+            if not picking and raise_if_not_found:
+                self._raise_create_batch_not_possible()
+            return picking
+        pickings = self._execute_search_pickings(domain)
         # at this stage we have the first picking to add to the batch but it could
         # exceed the limits of the available devices. In this case we split the
         # picking and return the picking to add to the batch. The split is done only
         # if the split_picking_exceeding_limits is set to True.
-        if (
-            picking
-            and self.split_picking_exceeding_limits
-            and self._is_picking_exceeding_limits(picking)
-        ):
-            split_picking = self._split_first_picking_for_limit(picking)
-            if not split_picking and raise_if_not_found:
-                raise PickingSplitNotPossibleError(picking)
-            picking = split_picking
-        return picking
+        selected_picking = self.env["stock.picking"]
+        for picking in pickings:
+            if self._is_picking_exceeding_limits(picking):
+                split_picking = self._split_first_picking_for_limit(picking)
+                if not split_picking:
+                    if raise_if_not_found:
+                        raise PickingSplitNotPossibleError(picking)
+                    _logger.warning(
+                        f"The picking {picking.name} could not be split "
+                        "for batch creation."
+                    )
+                    continue
+                selected_picking = split_picking
+            else:
+                selected_picking = picking
+            break
+        return selected_picking
```

Comment:
```suggestion
                return split_picking
            else:
                return picking
```

Reply:
If there is no picking in the `pickings`recordset, the function will return None instead of an empty recordset :frowning: